core/vm/instructions.go

```diff
@@ -670,9 +661,13 @@ func opCallCode(pc *uint64, evm *EVM, contract *Contract, memory *Memory, stack
 }
 
 func opDelegateCall(pc *uint64, evm *EVM, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
-	gas, to, inOffset, inSize, outOffset, outSize := stack.pop().Uint64(), stack.pop(), stack.pop(), stack.pop(), stack.pop(), stack.pop()
-
+	// Pop gas. The actual gas is in evm.callGasTemp.
+	stack.pop()
+	gas := evm.callGasTemp
+	// Pop other call parameters.
+	to, inOffset, inSize, retOffset, retSize := stack.pop(), stack.pop(), stack.pop(), stack.pop(), stack.pop()
```

Comment:
Perhaps rename `to` to `addr` so it's consistent with all the other call* changes?